cg/meta/clean/clean_flow_cells.py

```diff
@@ -0,0 +1,178 @@
+    ):
+        self.status_db: Store = status_db
+        self.hk_api: HousekeeperAPI = housekeeper_api
+        self.flow_cell = FlowCellDirectoryData(flow_cell_path=flow_cell_path)
+        self.current_time: float = time.time()
+        self.dry_run: bool = dry_run
+
+    def delete_flow_cell_directory(self) -> bool:
+        """
+        Delete the flow cell directory if it fulfills all requirements.
+        """
+        is_error_raised: bool = False
+        try:
+            self.set_sample_sheet_path_from_housekeeper()
+            if self.can_flow_cell_directory_be_deleted():
+                if self.dry_run:
+                    LOG.debug(f"Dry run: Would have removed: {self.flow_cell.path}")
+                    return is_error_raised
+                remove_directory_and_contents(self.flow_cell.path)
+                return is_error_raised
+        except Exception as error:
+            is_error_raised = True
+            LOG.error(f"Flow cell with path {self.flow_cell.path} not removed: {str(error)}")
+        return is_error_raised
+
+    def can_flow_cell_directory_be_deleted(self) -> bool:
+        """Determine whether a flow cell directory can be deleted."""
+        return all(
+            [
+                self.is_directory_older_than_21_days(),
+                self.is_flow_cell_in_statusdb(),
+                self.is_flow_cell_backed_up(),
+                self.has_sequencing_metrics_in_statusdb(),
+                self.has_fastq_files_for_samples_in_housekeeper(),
+                self.has_spring_meta_data_files_for_samples_in_housekeeper(),
+                self.has_spring_files_for_samples_in_housekeeper(),
+                self.has_sample_sheet_in_housekeeper(),
+            ]
+        )
+
+    def is_directory_older_than_21_days(self) -> bool:
+        """Check if a given directory is older than 21 days."""
+        return is_directory_older_than_days_old(
+            directory_path=self.flow_cell.path,
+            days_old=TWENTY_ONE_DAYS,
+            current_time=self.current_time,
+        )
+
+    def get_flow_cell_from_status_db(self) -> Optional[Flowcell]:
+        """
+        Get the flow cell entry from StatusDB.
+        Raises:
+            ValueError if the flow cell is not found in StatusDB.
+        """
+        flow_cell: Flowcell = self.status_db.get_flow_cell_by_name(self.flow_cell.id)
+        if not flow_cell:
+            raise ValueError(f"Flow cell {self.flow_cell.id} not found in StatusDB.")
+        return flow_cell
+
+    def is_flow_cell_in_statusdb(self) -> bool:
+        """Check if flow cell is in statusdb."""
+        return bool(self.get_flow_cell_from_status_db())
+
+    def is_flow_cell_backed_up(self) -> bool:
+        """Check if flow cell is backed up on PDC."""
+        return bool(self.get_flow_cell_from_status_db().has_backup)
+
+    def get_sequencing_metrics_for_flow_cell(self) -> Optional[List[SampleLaneSequencingMetrics]]:
```

Comment:
Are the metrics used in this class at all? What I see is that we only care about existence. Maybe this function and `has_sequencing_metrics_in_statusdb` can be combined